### PR TITLE
Fix intelligo adapter

### PR DESCRIPTION
**Pull request by cujomalainey**
API modification breakages
- fix create arg attributes
- fix buffer access helpers

### CODE REVIEW

File: src/audio/igo_nr/igo_nr.c [hunk 1]
```diff
@@ -402,7 +402,7 @@ static int32_t igo_nr_params(struct comp_dev *dev,
 	cd->source_rate = audio_stream_get_rate(&source_c->stream);
 	cd->sink_rate = audio_stream_get_rate(&sink_c->stream);
 
-	if (audio_stream_get_channels(source_c->stream) !=
+	if (audio_stream_get_channels(&source_c->stream) !=
```
Comment (btian1) [inline src/audio/igo_nr/igo_nr.c:405]: interesting, this part of code should not be compiled with current build system, compiler will report error with this cse.

**Review by btian1 — APPROVED** [review]
Are we try to re-integrate this part of code into SOF?

**Comment by cujomalainey** [comment]
> Are we try to re-integrate this part of code into SOF?

The code depends on a 3P library that hasn't been integrated past the CAVS2.5 branch. I'm working on adding stub libraries so we can at least build test and fuzz adapter code. These fixes are my findings from this work.

File: src/audio/igo_nr/igo_nr.c (hunk 1 above)
Comment (cujomalainey) [reply src/audio/igo_nr/igo_nr.c:405]: Yes, it's because it's not test in CI currently, trying to fix that :)

**Review by lyakh — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Thanks @cujomalainey ! A stub that would allow at least a compile test in CI would be nice to catch these indeed.

**Comment by kv2019i** [comment]
Lot of failing cases. We have a mismatch with upstream Zephyr main (not related to this PR), and we also have a regression in SOF main causing compiler warnings (again not related to this PR -> https://github.com/thesofproject/sof/pull/7949 ). So good to go for this PR.

**Pull request merged**